app/controllers/users_controller.rb

```diff
@@ -1,7 +1,50 @@
 class UsersController < ApplicationController
 
-  def list
+  def index
     @users = User.all
   end
-  alias :index :list
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    respond_to do |format|
+      if @user.save
+        format.html { redirect_to users_path, notice: "#{user.decorate.full_name} created"}
+      else
+        flash[:alert] = @user.errors.full_messages
+        format.html { render action: "new", status: :unprocessable_entity}
+      end
+    end
+  end
+
+  def destroy
+    if user.destroy
+      flash[:notice] = "#{user.decorate.full_name } deleted"
+      redirect_to root_path
+    end
+  end
+
+  # TODO: caching
+  def search
+    @users = User.search(search_params)
+  end
+
+  private
+
+  def user
```

Comment:
It's confusing when you use both ```@user``` and ```user``` in the plain in the same controller.  I recommend a before_action to clarify this and only use ```@user```.
```
before_action :find_user, only: :destroy
```